src/python/pants/engine/process.py

```diff
@@ -246,6 +336,52 @@ def try_decode(content: bytes) -> str:
         super().__init__("\n".join(err_strings))
 
 
+@rule
+async def run_maybe_coalesced_process_batch(
+    request: MaybeCoalescedProcessBatch,
+    use_coalesced_process_batch: ExperimentalCoalescedProcessBatchingOption,
+) -> FallibleProcessResult:
```

Comment:
I'm actually thinking this might be feasible as a `@rule_helper` and we strike it just right, can "do the right thing". E.g. do the normal mega-`await` in the normal case, or the mini-awaits in the split case.